### PR TITLE
[MDB IGNORE] Makes snowfort multi-z

### PR DESCRIPTION
**Pull request by ghost**
Idea somewhat based on Icebox Station from TG, but modified heavily

## Upper:
image
## Lower:
image
## Finer details:
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image

[general]
:cl:
 * rscadd: Snowfort Station has been renamed Snowbox Station, and has been given multi-z support, replacing the mining z-level with a lower mining area. The map can only be played on certain months if the station average temperature is below 0C at the end of a round.

### CODE REVIEW

**Comment by Kurfursten** [comment]
Some QoL you can bring in from Snaxi

Blacklist these events:
`event_blacklist = list(/datum/event/radiation_storm,/datum/event/carp_migration,/datum/event/rogue_drone,/datum/event/immovable_rod,/datum/event/meteor_wave,/datum/event/meteor_shower,/datum/event/thing_storm/meaty_gore,/datum/event/thing_storm/blob_shower,/datum/event/thing_storm/blob_storm,/datum/event/thing_storm/fireworks)`

Add these settings
`snow_theme = TRUE` - causes Christmas lobby screen and some minimap stuff for example showing trees
`can_enlarge = FALSE` - currently you have the spawn-only-exclusive-vaults setting because otherwise space vaults would stretch Z1 and create space on it, but this would prevent that

You might choose to generate snow vaults on this map, though you would need to tweak the logic somewhat for area spawning, for this call `generate_mapvaults()` in the map specific init (but again this one is less plug and play)
You may choose to add geyser spawning
You may want to map in some snowmobiles around exits

You can add these to disable blob and give people a snow guide:
```
/datum/map/active/map_ruleset(var/datum/dynamic_ruleset/DR)
	if(ispath(DR.role_category,/datum/role/blob_overmind))
		return FALSE

	return ..()

/datum/map/active/map_equip(var/mob/living/carbon/human/H)
	if(!istype(H))
		return
	H.equip_or_collect(new /obj/item/weapon/book/manual/snow(H.back), slot_in_backpack)
```

You can add this override to enable daynight
```
/datum/subsystem/daynightcycle
	flags = SS_FIRE_IN_LOBBY
	daynight_z_lvl = STATION_Z
```

To add a climate, add this to map specific init:
```
/datum/map/active/map_specific_init()
	climate = new /datum/climate/arctic()
```

Currently, you will walk off the edge of Z1 into space, which is terrifying. You can change this by using a snowsurface z level instead of a snow z level.

**Comment by Eneocho** [comment]
While very cool I don't see the map as the render shows it being used much, like 95% of the commonly used areas are in the upper level, you have almost no reason to go down to the lower one. There's almost no difference between it and just going to the mining shuttle.

File: maps/tgstation-snow.dm [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 	tDomeX = 128
 	tDomeY = 58
-	tDomeZ = 2
+	tDomeZ = 3
+	multiz = TRUE
+	height = 2
+	zLoc = 2
 	zLevels = list(
+		/datum/zLevel/snowmine,
 		/datum/zLevel/snow{
```
Comment (Kurfursten) [inline maps/tgstation-snow.dm:18]: what I meant is you want to use zLevel/snowsurface which is transition looped

Comment (ghost) [reply maps/tgstation-snow.dm:18]: The other zlevels are snow too, so there's no accidentally going into space

**Comment by AffectedArc07** [comment]
Requesting you put [MDB IGNORE] in the title. This crashed mapdiffbot twice this morning. 

You can render on final commit but not until then please.

**Comment by Kurfursten** [comment]
This image is not zoomed in enough for a serious review

**Comment by Kurfursten** [comment]
Post zoom ins of everywhere with open space, and also the frozen core area

**Comment by ghost** [comment]
That should cover most areas @Kurfursten

**Comment by Kurfursten** [comment]
I don't think borgs can use ladders.

**Comment by ghost** [comment]
Fixed up the borg spawn area, removed ladders and some other stuff @Kurfursten

**Comment by Kurfursten** [comment]
gotta take out those advanced cam computers as they are no longer a thing

**Pull request merged**